lib/systems/default.nix

```diff
@@ -112,15 +131,24 @@ let
         else if final.isNone                  then "newlib"
         # TODO(@Ericson2314) think more about other operating systems
         else                                     "native/impure";
+
+      unwinderlib =
+        /**/ if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "libunwind"
+        else "libgcc_s";
+
+      rtlib =
+        /**/ if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "compiler-rt"
+        else "libgcc";
+
       # Choose what linker we wish to use by default. Someday we might also
       # choose the C compiler, runtime library, C++ standard library, etc. in
       # this way, nice and orthogonally, and deprecate `useLLVM`. But due to
       # the monolithic GCC build we cannot actually make those choices
       # independently, so we are just doing `linker` and keeping `useLLVM` for
       # now.
       linker =
-        /**/ if final.useLLVM or false      then "lld"
-        else if final.isDarwin              then "cctools"
+        /**/ if final.useLLVM then "lld"
+        else if final.isDarwin then "cctools"
```

Comment:
I still think Darwin should be `ld64` not `cctools`, but that would require fixing a few references in-tree and potentially breaking external users.

Reply:
Yeah, I also would want it to be a later change since I don't want to push this to staging and having to wait for staging.